enumerable.rb

```diff
@@ -1,72 +1,87 @@
-module Enumerable
-    def my_each(arr)
-result=[]
-for i in 0..length
-    result.push(yield(arr[i]))
-end
-result
-    end
- 
-    def my_each_with_index(arr)
-        result = []
-        for i in 0..length
-            result.push(yield(arr[i],i))
-        end
-        result
-    end
-
-    def my_select
-        result=[]
-        my_each do |x|
-            output.push(x) if yield x
-        end
-        result
-    end
-
-    def my_all
-        result = true
-        my_each do |x|
-            result = false unless yield x
-        end
-        result
-    end
-
-    def my_any?
-        result = false
-        my_each do |x|
-            result = false if yield x
-        end 
-        result
-    end
-
-    def my_none?
-        result = true
-        my_each do |x|
-            result = false if yield x
-        end
-        result
-    end
-
-    def my_count
-        count = 0
-        my_each do |x|
-            count +=1 if yield x
-        end
-        count
-    end
-
-    def my_map(arr)
-        result = []
-        my_each(arr) do |x|
-            result.push(yield(x))
-        end
-        result
-    end
-
-
-
-
-
-
-
-end
\ No newline at end of file
+module Enumerable
+  def my_each(arr)
+    result = []
+    for i in 0..length
+      result.push(yield(arr[i]))
+    end
+    result
+  end
+
+  def my_each_with_index(arr)
+    result = []
+    for i in 0..length
+      result.push(yield(arr[i], i))
+    end
+    result
+  end
+
+  def my_select
+    result = []
+    my_each do |x|
+      output.push(x) if yield x
+    end
+    result
+  end
+
+  def my_all
```

Comment:
`my_all`, `my_none?`, `my_any?` and `my_count` should all be able to accept an argument.

These arguments can be a `Class`, `Regex`, etc


I would highly recommend checking the way they work on the [ruby doc](https://ruby-doc.org/core-2.6.5/Enumerable.html).

Also, investigate the use of `===` (`CaseEquality`) to make it easier to implement.

[Optional]

I would recommend using `my_all?` instead of `my_all` as the method return a boolean value (true or false)